### PR TITLE
[Abandoned]#162163282 Users can report Articles that violate terms

### PR DESCRIPTION
**Pull request by kafuuma**
#### What does this PR do?
This PR delivers a feature for users to report articles that violate terms 

#### Description of the tasks to be completed?
- A logged in user can report articles
- An article cannot be reported more than once
- Only Admin/super users can view a list of reported articles
- When an article is reported, it is immediately removed from published articles

#### End points workin
```
- POST "/api/articles/<str:slug>/report/"
- GET "/api/reported/"
```

#### How should this be manually tested?

```
- Clone repository
- create a virtual environment
- Install requirements `pip install -r requirements.txt`
- Run the project `python manage.py runserver`
```
```
Paste URL in postman: `http://localhost:8000/api/articles/<str:slug>/report/`
```
```
Report body
```
```
{
	"report":{
		"reason":"this article has been plagiarised"
	}
}
```

Screenshots 

1. Report an article

rep1

2. List reported articles

report

#### What are the relevant pivotal tracker stories

[#162163282](https://www.pivotaltracker.com/story/show/162163282)

### CODE REVIEW

File: authors/apps/articles/renderers.py [hunk 1]
```diff
@@ -38,3 +38,33 @@ def render(self, data, media_type=None, renderer_context=None):
...
+class ArticleListReportJSONRenderer(JSONRenderer):
+    charset = 'utf-8'
+
+    def render(self, data, media_type=None, renderer_context=None):
+
+        errors = data.get('errors', None)
+
+        if errors is not None:
+
+            return super(ArticleListReportJSONRenderer, self).render(data)
```
Comment (Kasulejoseph) [inline authors/apps/articles/renderers.py:66]: henry, you missed testing this renderer class

**Review by Kasulejoseph — APPROVED** [review]

File: authors/apps/articles/renderers.py (hunk 1 above)
Comment (kafuuma) [reply authors/apps/articles/renderers.py:66]: I get but I tested almost  every functionality
But that line remained, however, my coverage passes

**Comment by Kasulejoseph** [comment]
> #### What does this PR do?
> This PR delivers a feature for users to report articles that violate terms
> 
> #### Description of the tasks to be completed?
> * A logged in user can report articles
> * An article cannot be reported more than once
> * Only Admin/super users can view a list of reported articles
> * When an article is reported, it is immediately removed from published articles
> 
> #### End points workin
> ```
> - POST "/api/articles/<str:slug>/report/"
> - GET "/api/reported/"
> ```
> #### How should this be manually tested?
> ```
> - Clone repository
> - create a virtual environment
> - Install requirements `pip install -r requirements.txt`
> - Run the project `python manage.py runserver`
> ```
> ```
> Paste URL in postman: `http://localhost:8000/api/articles/<str:slug>/report/`
> ```
> ```
> Report body
> ```
> ```
> {
> 	"report":{
> 		"reason":"this article has been plagiarised"
> 	}
> }
> ```
> Screenshots
> 
> 1. Report an article
> 
> rep1
> 
> 1. List reported articles
> 
> report
> 
> #### What are the relevant pivotal tracker stories
> [Delivers]#162163282]

story id is just enough, remove the `[Delivers]`

**Review by SokoPaulSokool — APPROVED** [review]

**Review by bibangamba — APPROVED** [review]

**Review by collin5 — CHANGES_REQUESTED** [review]
I think reporting should not automatically remove an article, It should be the admin to remove the article after verifying that this is not a false positive.
Admin should also be able to dismiss false positive reports

**Pull request closed without merging**